src/execution/compiler/operator/hash_aggregation_translator.cpp

```diff
@@ -29,6 +29,19 @@ HashAggregationTranslator::HashAggregationTranslator(const planner::AggregatePla
   TERRIER_ASSERT(plan.GetAggregateStrategyType() == planner::AggregateStrategyType::HASH,
                  "Expected hash-based aggregation plan node");
   TERRIER_ASSERT(plan.GetChildrenSize() == 1, "Hash aggregations should only have one child");
+  // TODO(kunal&ricky):
```

Comment:
Since _sum_ and _count_ are both static aggregations, using a hash table in `static_aggregation_translator.cpp` seems sufficient to me for using distinct with either of these aggregations. I don't think editing `hash_aggregation_translator.cpp` is necessary. Thoughts?

Reply:
I think for the `hash_aggregation_translator`, it is now handling cases with presence of Group By clauses. Consider a query like `SELECT COUNT(DISTINCT x), y FROM aaa GROUP BY y`, it will also need to do the distinct aggregate rather than the normal aggregate.